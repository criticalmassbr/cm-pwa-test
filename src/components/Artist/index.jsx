import React from 'react';

import {
  Container,
  ArtistImage,
  ArtistName,
} from './styles';

function Artist ({ artist }) {
  return (
    <Container>
      <ArtistImage src={ artist.image } />
      <ArtistName>
        { artist.name }
      </ArtistName>
    </Container>
  );
}

export default Artist;
