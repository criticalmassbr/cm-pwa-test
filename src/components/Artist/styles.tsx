import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  width: 40%;
  margin: 0 auto;
  padding: 1.5em 0;

  @media (max-width: 767px) {
    width: 50%;
  }
`;

export const Box = styled.div`
  flex: 1 0 50%;
`;

export const Title = styled(Box)`
  text-align: center;
  font-size: 1.25em;
  font-weight: bold;
`;

export const Cover = styled.img`
  width: 80%;
  height: auto;
  border-radius: 10em;

  @media (max-width: 767px) {
    width: 100%;
  }
`;
