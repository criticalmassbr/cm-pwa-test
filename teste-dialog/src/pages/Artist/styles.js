import styled from "styled-components";

export const Container = styled.div``;

export const Text = styled.text`
  color: #fff;
  font-family: sans-serif;
  margin-top: 20px;
  text-align: center;
  justify-content: center;
  display: flex;
  font-size: 30px;
`;
