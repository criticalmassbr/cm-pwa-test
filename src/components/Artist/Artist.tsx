import React from 'react';
import { Box, Container, Cover, Title } from './styles';

type Props = {
  image: string;
  name: string;
};
function Artist({ image, name }: Props) {
  return (
    <Container>
      {image && (
        <Box>
          <Cover src={image} alt="Artist" />
        </Box>
      )}
      {name && <Title title="artist name">{name}</Title>}
    </Container>
  );
}

export default Artist;
