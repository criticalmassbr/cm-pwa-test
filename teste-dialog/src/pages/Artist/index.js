import React from "react";

import Search from "../../Components/Search";
import PhotoArtist from "../../Components/PhotoArtist";
import ListMusics from "../../Components/ListMusics";

import { Container } from "./styles";

export default function Artist() {
  return (
    <Container>
      <Search />
      <PhotoArtist />
      <ListMusics />
    </Container>
  );
}
