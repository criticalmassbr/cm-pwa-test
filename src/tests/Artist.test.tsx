import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Artist from '../components/Artist/Artist';

test('renders artist image and name', () => {
  const component = <Artist name="Test" image="teste.jpg" />;
  const artistComponent = renderer.create(component);
  const tree = artistComponent.toJSON();
  expect(tree).toMatchSnapshot();

  const renderResult = render(component);
  const artistImage = renderResult.getByAltText(/artist/i);
  expect(artistImage).toBeInTheDocument();
  const artistName = renderResult.getByTitle(/artist name/i);
  expect(artistName).toBeInTheDocument();
});
