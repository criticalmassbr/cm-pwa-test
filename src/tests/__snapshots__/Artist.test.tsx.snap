// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders artist image and name 1`] = `
<div
  className="sc-AxjAm UcREr"
>
  <div
    className="sc-AxirZ dRbqjZ"
  >
    <img
      alt="Artist"
      className="sc-AxhCb fCPykX"
      src="teste.jpg"
    />
  </div>
  <div
    className="sc-AxirZ sc-AxiKw dlHzCf"
    title="artist name"
  >
    Test
  </div>
</div>
`;
